src/wallet/rpc/backup.cpp

```diff
@@ -93,6 +93,21 @@ static void RescanWallet(CWallet& wallet, const WalletRescanReserver& reserver,
     }
 }
 
+static void EnsureBlockDataFromTime(interfaces::Chain& chain, const CWallet& wallet, int64_t timestamp)
+{
+    if (!chain.havePruned())
+        return;
+
+    int height{0};
+    const bool found{chain.findFirstBlockWithTimeAndHeight(timestamp - TIMESTAMP_WINDOW, 0, FoundBlock().height(height))};
+    const std::optional<int> chainHeight{chain.getHeight()};
```

Comment:
In commit "rpc: Enable wallet import on pruned nodes and add test" (f42c077a8f948f5c93b68eb1b7fa246b4f6475ef)

I think there's no need for the chainHeight variable anymore and it would be better to drop it.